poetry/semver/version.py

```diff
@@ -197,6 +197,11 @@ def include_max(self):
     @classmethod
     def parse(cls, text):  # type: (str) -> Version
         match = COMPLETE_VERSION.match(text)
```

Comment:
You'll need to delete this line first, it seems

Reply:
Wait why? It's only failing in Python 2.7

Reply:
What I mean to say is that, line 199 `match = COMPLETE_VERSION.match(text)` is the same as line 201 inside the `try`/`catch`.

So when `.match` raises a `TypeError` it never makes it to the `try`/`catch`ed version.

I deleted line 199, and it appeared to fix https://github.com/sdispater/poetry/issues/837